### PR TITLE
Add support for Python 3.10

### PR DESCRIPTION
**Pull request by yurnih**
This PR adds Python 3.10 to the CI pipeline. Furthermore, it also fixes some failing tests which showed up by upgrading `pytest`.
_Update: Branch renamed due to failure in ci, but still fails during stage "Autoformat code" because of git-related error._

### CODE REVIEW

**Comment by carloslubbers** [comment]
Thanks @yurnih , I've fixed the git related issue in the master branch. If you merge it into your branch it should pass. We still have some issues on macos builds which we will look at later, can be ignored for now if you encounter them

**Comment by yurnih** [comment]
Merged master into PR. The fix didn't solved the issue with git, but it works in ci of own fork: https://github.com/yurnih/GEOLib/runs/7087359531.

**Comment by carloslubbers** [comment]
I see, our pipeline may not work well with external PR's yet, I will discuss this with the other maintainers, you can ignore the error for now!

File: geolib/models/dstability/internal.py [hunk 1]
```diff
@@ -813,6 +813,10 @@ def edit_soil(self, code: str, **kwargs: dict) -> PersistableSoil:
                 for k, v in kwargs.items():
                     try:
                         setattr(persistable_soil, snake_to_camel(k), v)
+
+                        k_stochastic = f"{snake_to_camel(k)}StochasticParameter"
```
[the remark this answers is not in the document]
Comment (yurnih) [reply geolib/models/dstability/internal.py:817]: The issue here is that by adding the soil to the d-stability model, by default a Stochastic_parameter.mean is used. However, by reversing with `model.get_soil`, the model makes a new soil model, based on the values of the internal soil model. During this conversion, the parameters are only based on 

https://github.com/Deltares/GEOLib/blob/fd66ecdc01f90da1fa4383791e6881f16fcdbb1c/geolib/models/dstability/internal.py#L737-L748

When not setting the `StochasticParameter`-part, the result of the `model.get_soil` will never receive the updated values and will always fail at 
https://github.com/Deltares/GEOLib/blob/fd66ecdc01f90da1fa4383791e6881f16fcdbb1c/tests/models/dstability/test_dstability_soils.py#L100-L116

[the remark this answers is not in the document]
Comment (carloslubbers) [reply geolib/models/dstability/internal.py:817]: Thanks,  I understand now! It was a little confusing the Python 3.10 support was mixed with bug fixes as well. Don't be afraid to open multiple PR's for bug fixes in the future, it makes reviewing a lot easier!

**Review by carloslubbers — APPROVED** [review]

**Pull request merged**